data/src/main/java/com/microsoft/azure/kusto/data/exceptions/DataClientException.java

```diff
@@ -22,4 +25,8 @@ public DataClientException(String ingestionSource, String message, Exception exc
         super(message, exception);
         this.ingestionSource = ingestionSource;
     }
+
+    public Boolean isPermanent(){
```

Comment:
Not in scope for this change, but ideally we'd have a parent interface that defines isPermanent, and the Client and Service exceptions would both inherit and provide their implementations.